app/controllers/github_webhooks_controller.rb

```diff
@@ -0,0 +1,36 @@
+class GithubWebhooksController < ApplicationController
+  
+  # POST /author_webhook
+  def author_webhook
+    action = params["github_webhook"]["action"]
+    case action
+    when "opened"
+      create_author(params["issue"])
+    when "edited"
+      update_author(params["issue"])
+    when "deleted"
+      delete_author(params["issue"])
+    end
+  end
+
+  def create_author(issue)
+    author = Author.create(name: issue["title"], biography: issue["body"], github_id: issue["number"])
+    author.books.create(title: Faker::Book.title, publisher: author)
+  end
+
+  def update_author(issue)
+    author = find_author(issue["number"])
+    author.update_attributes(name: issue["title"], biography: issue["body"]) if author
+  end
+
+  def delete_author(issue)
+    author = find_author(issue["number"])
+    author.destroy if author
+  end
+
+  private
+
+  def find_author(id)
+    Author.find_by_github_id(id)
```

Comment:
These `find_by_x` methods are not as performant as the equivalent `find_by(x: y)` version, since the `find_by_x` version relies on `method_missing`. Please use `find_by` instead.

Reply:
Update changes and noted for next time as well
Thanks